core/src/main/scala/org/broadinstitute/dsde/rawls/openam/StandardUserInfoDirectives.scala

```diff
@@ -1,23 +1,52 @@
 package org.broadinstitute.dsde.rawls.openam
 
-import org.broadinstitute.dsde.rawls.model.{RawlsUserEmail, RawlsUserSubjectId, UserInfo}
+import org.broadinstitute.dsde.rawls.dataaccess.{HttpSamDAO, SamDAO, SlickDataSource}
+import org.broadinstitute.dsde.rawls.dataaccess.jndi.JndiDirectoryDAO
+import org.broadinstitute.dsde.rawls.model.{RawlsUser, RawlsUserEmail, RawlsUserRef, RawlsUserSubjectId, UserInfo, UserStatus}
 import spray.http.{HttpHeader, OAuth2BearerToken}
 import spray.routing.Directive1
 import spray.routing.Directives._
 
-trait StandardUserInfoDirectives extends UserInfoDirectives {
+import scala.concurrent.Future
 
+trait StandardUserInfoDirectives extends UserInfoDirectives with JndiDirectoryDAO {
+
+  val httpSamDAO: SamDAO
+
+  //TODO: project should be regex
+  val petSAdomain = "\\S+@\\S+.iam.gserviceaccount.com"
+  val petEmailPrefix = "pet-"
   def requireUserInfo(): Directive1[UserInfo] = {
-    for(accessToken <- accessTokenHeaderDirective;
-        userEmail <- emailHeaderDirective;
-        accessTokenExpiresIn <- accessTokenExpiresInHeaderDirective;
-        userSubjectId <- userSubjectIdDirective
+
+    val userInfo = for(
```

Comment:
I know this was like this before, but can we change this for comprehension to:
```
    val userInfo = for {
      accessToken <- accessTokenHeaderDirective
      userEmail <- emailHeaderDirective
      accessTokenExpiresIn <- accessTokenExpiresInHeaderDirective
      userSubjectId <- userSubjectIdDirective
    } yield UserInfo(RawlsUserEmail(userEmail), OAuth2BearerToken(accessToken), accessTokenExpiresIn.toLong, RawlsUserSubjectId(userSubjectId))
```
?

Reply:
Is one better than the other?

Reply:
Semicolons aren't really idiomatic Scala. It's just a style thing.

Reply:
yeah ^